camptix.php

```diff
@@ -1900,21 +1900,36 @@ function sort_currencies( $a, $b ) {
 	 */
 	function append_currency( $amount, $nbsp = true, $currency_key = false ) {
 		$currencies = $this->get_currencies();
-		$currency = $currencies[ $this->options['currency'] ];
-		if ( $currency_key )
-			$currency = $currencies[ $currency_key ];
 
-		if ( ! $currency )
-			$currency = array( 'label' => __( 'U.S. Dollar', 'camptix' ), 'locale' => 'en_US.UTF-8' );
-
-		if ( isset( $currency['locale'] ) ) {
-			setlocale( LC_MONETARY, $currency['locale'] );
+		if ( ! $currency_key ) {
+			if ( isset( $this->options[ 'currency' ] ) ) {
+				$currency_key = $this->options['currency'];
+			} else {
+				$currency_key = 'USD' ;
+			}
 		}
 
-		$formatted_amount = money_format( '%n', $amount );
+		$currency = $currencies[ $currency_key ];
 
-		if ( isset( $currency['format'] ) && $currency['format'] ) {
+		if ( isset( $currency['locale'] ) ) {
+			// money_format not available on Windows and some other systems
+			if ( function_exists( 'money_format' ) ) {
+				setlocale( LC_MONETARY, $currency['locale'] );
+				$formatted_amount = money_format( '%n', $amount );
+			} else {
+				/**
+				 * set formatted amount manually, because if `locale` is defined for currency then `format` may not be
+				 * defined
+				 */
+				$formatted_amount = ( $currency_key ? $currency_key . ' ' : '' ) . $amount;
```

Comment:
Is the ternary operator necessary here? `$currency_key` will always have some value, as of line 1908, right?

If so, then this line will be the same as 1931, so it looks like we might be able to simplify the conditionals a bit. Something like:

```php
if ( isset( $currency['locale'] ) && function_exists( 'money_format' ) ) {
	setlocale( LC_MONETARY, $currency['locale'] );
	$formatted_amount = money_format( '%n', $amount );
} elseif ( isset( $currency['format'] ) && $currency['format'] ) {
	$formatted_amount = sprintf( $currency['format'], number_format( (float) $amount, 2 ) );
} else {
	$formatted_amount = $currency_key . ' ' . $amount;
}
```

Does that seem right, or am I missing something?

Reply:
Yup you are correct. Made the change.